pandas/core/frame.py

```diff
@@ -1336,7 +1343,15 @@ def to_numpy(self, dtype=None, copy: bool = False) -> np.ndarray:
         array([[1, 3.0, Timestamp('2000-01-01 00:00:00')],
                [2, 4.5, Timestamp('2000-01-02 00:00:00')]], dtype=object)
         """
-        result = np.array(self.values, dtype=dtype, copy=copy)
+        result = self._mgr.as_array(
+            transpose=self._AXIS_REVERSED, dtype=dtype, na_value=na_value
+        )
+
+        if copy or result.dtype is not dtype:
+            # FIXME: The dtype conversion is potentially needed since the
+            #  single extension block case is not correctly cast above
+            result = np.array(result, dtype=dtype, copy=copy)
```

Comment:
Not happy with this dtype check, but otherwise there's a regression in the single extension block case. Open to ideas / help on why we can't just do this via to_numpy around where we call get_values in as_array (several tests in frame/methods/test_append.py fail for instance)

Reply:
yeah this should be handled in .as_array itself